_unit-test/error-handling-test.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+set -ex
+
+source "$(dirname $0)/../install/_lib.sh"
+
+export REPORT_SELF_HOSTED_ISSUES=1
+
+source ../install/error-handling.sh
+
+send_envelope() {
+  echo "Test Sending $1"
+}
+
+export -f send_envelope
+echo "${_group}Testing initial send_event"
+export log_file="test_log.txt"
+echo "Test Logs" > "$basedir/$log_file"
+SEND_EVENT_RESPONSE=$(send_event "12345123451234512345123451234512" Test exited with status 1)
+if [ "$SEND_EVENT_RESPONSE" != 'Test Sending sentry-envelope-12345123451234512345123451234512' ]; then
+  exit -1
+fi
+ENVELOPE_CONTENTS=$(cat /tmp/sentry-envelope-12345123451234512345123451234512)
+if [ "$ENVELOPE_CONTENTS" !='{"event_id":"12345123451234512345123451234512","dsn":"https://19555c489ded4769978daae92f2346ca@self-hosted.getsentry.net/3"}
```

Comment:
I think it might be better to have expected envelopes as files we can cat.

Reply:
done!